### DISCUSSION TITLE
Rotate 2D Graph?

### DISCUSSION THREAD

**Discussion by rsimon** [vasturiano/react-force-graph#523] [Q&A] [answered]
Hi, first off: thanks for such a fantastic library!

I have a feature question: is there a way to rotate the 2D graph? (Eventually, I'd love to to be able to rotate the graph canvas by dragging the mouse. But the mouse handling I can easily build myself - as long as there's a way to programmatically control the world rotation angle.)

**Comment by vasturiano** [comment, marked as answer]
@rsimon thanks for reaching out.

There isn't a built-in way to rotate the canvas itself, but you could achieve this by simply changing the x,y coordinates of the nodes themselves, based on your mouse handling events. For each of the nodes you can access and modify their current coordinates by manipulating the `x` and `y` attributes of each node object.

**Reply by rsimon** [reply]
Thanks for the quick reply. I wasn't aware I could just change the coordinates (without breaking things ;-). I'll brush up my trigonometry and will give it a try!
